File: chapter02/sanghyo/MovieReservationProgram/MovieReservationProgram/SequenceCondition.swift
//
//  SequenceCondition.swift
//  MovieReservationProgram
//
//  Created by sanghyo on 11/29/23.
//

import Foundation

struct SequenceCondition: DiscountCondition {
    private let sequence: Int
    
    init(sequence: Int) {
        self.sequence = sequence
    }
 
    func isSatisfiedBy(screening: Screening) -> Bool {
        return screening.isSequence(sequence: sequence)
    }
}

File: chapter02/hm4725/src/SequenceCondition.cc
#include "SequenceCondition.hh"
#include "Screening.hh"

SequenceCondition::SequenceCondition(int sequence) : sequence(sequence) {}
bool SequenceCondition::isSatisfiedBy(std::shared_ptr<Screening> screening) {
  return screening->isSequence(sequence);
}